libraries/ESP8266WiFi/src/ESP8266WiFiScan.cpp

```diff
@@ -147,6 +147,14 @@ void ESP8266WiFiScanClass::scanDelete() {
     _scanComplete = false;
 }
 
+/**
+ * returns const pointer to the requested scanned wifi entry for furthor parsing.
+ * @param networkItem uint8_t
+ * @return struct bss_info*, may be NULL
+ */
+const struct bss_info *ESP8266WiFiScanClass::getNetworkInfo(uint8_t i) {
+    return reinterpret_cast<const struct bss_info*>(_getScanInfoByIndex(i));
+}

```

Comment:
should we match the current esp32 api by name?
https://github.com/espressif/arduino-esp32/blob/a5f03a86512b270be1080d7e7cb82e4fd71c9a1a/libraries/WiFi/src/WiFiScan.h#L48
imo I'd prefer to still use a hard type. I don't see much benefit of either ambiguous void or creating our own type for whatever (...encapsulation?) reason. but, whatever approach works

minor nit: I suppose the idea is to refer to global bss info type, but we can totally lose the 'struct' since c++ does not care about that and it could be just `const bss_info*`. plus we will see definite warning if we ever move headers and forget user interface, `struct bss_info*` will just warn about something undefined instead of hard-failing

Reply:
Thanks, I meant to look at how the ESP32 API did it and forgot. I'll do an update.